### ISSUE TITLE
Study & Explain the SVD feature

### ISSUE THREAD

**Issue by alexwalterbos** [alexwalterbos/nlp_fake_news#6]
The way Talos explained the SVD feature module is "This module takes the TF-IDF features and applies Singular-Value Decomposition to them to obtain a compact, dense vector representation of the headline and body respectively. This procedure is well known and corresponds to finding the latent topics involved in the corpus and represent each headline/body text as a mixture of these topics. The cosine similarities between the SVD features of headline and body text are also computed. This similarity metric is very indicative of whether the body is related to the headline or not." 

The well known procedure is the latent semantic analysis procedure.  Latent Semantic Analysis is a technique for creating a vector representation of a document. A small example from [https://technowiki.wordpress.com/2011/08/27/latent-semantic-analysis-lsa-tutorial/](url) (not for in the report, but to understand):  

> As a small example, I searched for books using the word “investing” at Amazon.com and took the top 10 book titles that appeared. One of these titles was dropped because it had only one index word in common with the other titles. An index word is any word that:
> 
> appears in 2 or more titles, and
> is not a very common word such as “and”, “the”, and so on (known as stop words). These words are not included because do not contribute much (if any) meaning.
> In this example we have removed the following stop words: “and”, “edition”, “for”, “in”, “little”, “of”, “the”, “to”.
> 
> Here are the 9 remaining tiles. The index words (words that appear in 2 or more titles and are not stop words) are underlined.
> 
> The Neatest Little Guide to Stock Market Investing
> Investing For Dummies, 4th Edition
> The Little Book of Common Sense Investing: The Only Way to Guarantee Your Fair Share of Stock Market Returns
> The Little Book of Value Investing
> Value Investing: From Graham to Buffett and Beyond
> Rich Dad’s Guide to Investing: What the Rich Invest in, That the Poor and the Middle Class Do Not!
> Investing in Real Estate, 5th Edition
> Stock Investing For Dummies
> Rich Dad’s Advisors: The ABC’s of Real Estate Investing: The Secrets of Finding Hidden Profits Most Investors Miss
> 
> Once Latent Semantic Analysis has been run on this example, we can plot the index words and titles on an XY graph and identify clusters of titles. The 9 titles are plotted with blue circles and the 11 index words are plotted with red squares. Not only can we spot clusters of titles, but since index words can be plotted along with titles, we can label the clusters. For example, the blue cluster, containing titles T7 and T9, is about real estate. The green cluster, with titles T2, T4, T5, and T8, is about value investing, and finally the red cluster, with titles T1 and T3, is about the stock market. The T6 title is an outlier, off on its own.
In short, LSA uses SVD to perform dimensionality reduction on the TF-IDF vectors.
